File: ScriptableObjectModel/ScriptableObjectModel/ScriptingLanguage.cs

namespace ScriptableObjectModel
{
    [Scriptable] public static class ScriptingLanguage
    {
        internal static IDrawing Drawing { get; private set; }

        public static void SetDrawing(IDrawing drawing)
        {
            Drawing = drawing;
        }

        [Scriptable] public static int PictureWidth { get { return Drawing.PictureWidth; } }
        [Scriptable] public static int PictureHeight { get { return Drawing.PictureHeight; } }

        [Scriptable] public static void DrawLine(int x1, int y1, int x2, int y2)
        {
            Drawing.DrawLine(x1, y1, x2, y2);
        }

        [Scriptable] public static void EraseLines()
        {
            Drawing.EraseLines();
        }
    }
}

File: VBScriptableObjectModel/VBScriptableObjectModel/ScriptingLanguage.vb
<Scriptable> Public Module ScriptingLanguage
    Private _IDrawing As IDrawing

    Public Sub Initialize(aIDrawing As IDrawing)
        _IDrawing = aIDrawing
    End Sub

    <Scriptable>
    Public ReadOnly Property PictureWidth() As Integer
        Get
            Return _IDrawing.PictureWidth
        End Get
    End Property

    <Scriptable>
    Public ReadOnly Property PictureHeight() As Integer
        Get
            Return _IDrawing.PictureHeight
        End Get
    End Property

    <Scriptable>
    Public Sub DrawLine(x1 As Integer, y1 As Integer, x2 As Integer, y2 As Integer)
        _IDrawing.DrawLine(x1, y1, x2, y2)
    End Sub

    <Scriptable>
    Public Sub EraseLines()
        _IDrawing.EraseLines()
    End Sub

End Module
